proof/crefine/ARM/Arch_C.thy

```diff
@@ -1885,6 +1883,132 @@ lemma performPageInvocationMapPDE_ccorres:
   apply (auto simp: unat_arith_simps unat_word_ariths)
   done
 
+lemma length_of_msgRegisters:
+  "length ARM_H.msgRegisters = 4"
+  by (auto simp: msgRegisters_unfold)
+
+lemma setMRs_single:
+  "setMRs thread buffer [val] = do y \<leftarrow> asUser thread (setRegister register.R2 val);
```

Comment:
```suggestion
  "setMRs thread buffer [val] = do _ \<leftarrow> asUser thread (setRegister register.R2 val);
```

Reply:
I actually think this didn't apply without binding to a variable, I'll have another look

Reply:
Like this is should still bind to a variable, it just doesn't specify a name (i.e. it's different from `do asUser thread (setRegister register.R2 val); .. od`, which is `K_bind` internally)